jupyterhub/proxy.py

```diff
@@ -61,8 +112,8 @@ def add_route(self, routespec, target, data):
         """Add a route to the proxy.
 
         Args:
-            routespec (str): A specification for which this route will be matched.
-                Could be either a url_prefix or a fqdn.
+            routespec (RouteSpec or str): A specification for which this route will be matched.
```

Comment:
When will we ever pass in a string? IMO proxy implementations should never have to care about casting into RouteSpec.